app/org/sagebionetworks/bridge/services/UserAdminService.java

```diff
@@ -132,14 +133,20 @@ public UserSession createUser(SignUp signUp, Study study, SubpopulationGuid subp
 
             if (consentUser) {
                 String name = String.format("[Signature for %s]", signUp.getEmail());
-                ConsentSignature consent = new ConsentSignature.Builder().withName(name)
+                ConsentSignature signature = new ConsentSignature.Builder().withName(name)
                         .withBirthdate("1989-08-19").withSignedOn(DateUtils.getCurrentMillisFromEpoch()).build();
-
-                SubpopulationGuid consentTo = (subpopGuid != null) ? subpopGuid
-                        : SubpopulationGuid.create(study.getIdentifier());
-
-                consentService.consentToResearch(study, consentTo,
-                        newUserSession.getUser(), consent, SharingScope.NO_SHARING, false);
+                
+                User user = newUserSession.getUser();
+                if (subpopGuid != null) {
+                    consentService.consentToResearch(study, subpopGuid, user, signature, NO_SHARING, false);
+                } else {
+                    for (ConsentStatus consentStatus : newUserSession.getUser().getConsentStatuses().values()) {
```

Comment:
Nitpick: You already have newUserSession.getUser() saved in var user, so you don't need to call it again here.